### DISCUSSION TITLE
Automating SQL4CDS Updates

### DISCUSSION THREAD

**Discussion by thgoddard** [MarkMpn/Sql4Cds#227] [Q&A]
I have a SQL script that runs successfully in SQL4CDS in SSMS, updating several records in Dataverse.  Works great - thanks!
However I have to run this script manually from SSMS, and I would like to automate its execution so it runs nightly.  What options are available or recommended for this?
Thanks - Tom

**Comment by MarkMpn** [comment]
Thanks Tom! I don't currently have anything ready-made for this, but it's something I'd like to build. So let me turn it around - what would work best for you? I could envisage a command-line utility, PowerShell script, Azure DevOps or GitHub Actions task, ...

For each of these, the scheduling component would be done by something else (Azure Pipelines, Windows Task Scheduler, SQL Server Agent etc.) but I could produce an automatable task that could be run by those systems.

**Reply by NickPl** [reply]
Hi Mark, I would also like this feature. So, I will answer for Tom.

I'm thinking about that feature a lot. My use cases are:
- Run a script every day to fix data problematic data
- Run a script pre or post deployment

I would also envisage a command-line utility using a inline command or an sql file. I'm currently using Azure DevOps & Powershell. We'll switch to GitHub in a year.

**Comment by SiPurdy** [comment]
Doing both aspects of this already.

- Using [dependent assemblies](https://learn.microsoft.com/en-us/power-apps/developer/data-platform/build-and-package#dependent-assemblies) for including the Sql4Cds assemblies for use in plugins. We have a nice system built up for running tests (defined by rows on a table) and applying fixes.
- For deployments, we use MS Deployment Packages so we just use straight references in our assembly for ms packages.
